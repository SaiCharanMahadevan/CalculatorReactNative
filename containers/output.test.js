import React from 'react';
import { shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components/native';

import THEME from 'constants/theme';
import OutputContainer from './output';

describe('OutputContainer', () => {
  test('renders the section that displays the result of the calculation', () => {
    const wrapper = shallow(
      <ThemeProvider theme={THEME}>
        <OutputContainer expression="1+2-3" result="0"/>
      </ThemeProvider>,
    ).dive().dive().dive();
    expect(wrapper).toMatchSnapshot();
  });
});
