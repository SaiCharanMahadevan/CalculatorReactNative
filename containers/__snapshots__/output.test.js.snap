// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutputContainer renders the section that displays the result of the calculation 1`] = `
<Styled(View)
  alignItems="flex-end"
  flex={3}
>
  <Styled(Styled(View))
    flex={6}
  >
    <Styled(Text)
      adjustsFontSizeToFit={true}
      size="large"
      weight="bold"
    >
      1+2-3
    </Styled(Text)>
  </Styled(Styled(View))>
  <Styled(View) />
  <Styled(Styled(View))
    flex={4}
  >
    <Styled(Text)
      adjustsFontSizeToFit={true}
      result={true}
      size="medium"
      weight="semibold"
    >
      0
    </Styled(Text)>
  </Styled(Styled(View))>
</Styled(View)>
`;
